google/datalab/contrib/bigquery/commands/_bigquery.py

```diff
@@ -0,0 +1,231 @@
+    context = google.datalab.bigquery.commands._bigquery._construct_context_for_args(args)
+    pipeline_spec = _get_pipeline_spec_from_config(bq_pipeline_config, args)
+    pipeline = google.datalab.contrib.pipeline._pipeline.Pipeline(name, pipeline_spec)
+    google.datalab.utils.commands.notebook_environment()[name] = pipeline
+
+    debug = args.get('debug')
+    if debug is True:
+      return pipeline.py
+
+
+def _get_pipeline_spec_from_config(bq_pipeline_config, cell_args):
+  input_config = bq_pipeline_config.get('input', None)
+  transformation_config = bq_pipeline_config.get('transformation', None)
+  output_config = bq_pipeline_config.get('output', None)
+
+  load_task_config_name = 'bq_pipeline_load_task'
+  execute_task_config_name = 'bq_pipeline_execute_task'
+  extract_task_config_name = 'bq_pipeline_extract_task'
+
+  load_task_config = _get_load_parameters(input_config)
+  execute_task_config = _get_execute_parameters(load_task_config_name, transformation_config,
+                                                cell_args)
+  extract_task_config = _get_extract_parameters(execute_task_config_name, execute_task_config,
+                                                output_config)
+
+  pipeline_spec = {
+    'email': bq_pipeline_config['email'],
+    'schedule': bq_pipeline_config['schedule'],
+  }
+
+  pipeline_spec['tasks'] = {}
+  if load_task_config:
+    pipeline_spec['tasks'][load_task_config_name] = load_task_config
+  if execute_task_config:
+    pipeline_spec['tasks'][execute_task_config_name] = execute_task_config
+  if extract_task_config:
+    pipeline_spec['tasks'][extract_task_config_name] = extract_task_config
+
+  if not load_task_config and not execute_task_config and not extract_task_config:
+    raise Exception('Pipeline has no tasks to execute.')
+
+  return pipeline_spec
+
+
+def _get_load_parameters(bq_pipeline_input_config):
+    load_task_config = {'type': 'pydatalab.bq.load'}
+    path_exists = False
+    if 'path' in bq_pipeline_input_config:
+      # The path URL of the GCS load file(s).
+      load_task_config['path'] = bq_pipeline_input_config['path']
+      path_exists = True
+
+    table_exists = False
+    if 'table' in bq_pipeline_input_config:
+      # The destination bigquery table name for loading
+      load_task_config['table'] = bq_pipeline_input_config['table']
+      table_exists = True
+
+    schema_exists = False
+    if 'schema' in bq_pipeline_input_config:
+      # The schema of the destination bigquery table
+      load_task_config['schema'] = bq_pipeline_input_config['schema']
+      schema_exists = True
+
+    # We now figure out whether a load operation is required
+    if table_exists:
+      if path_exists:
+        if schema_exists:
+          # One of 'create' (default), 'append' or 'overwrite' for loading data into BigQuery. If a
+          # schema is specified, we assume that the table needs to be created.
+          load_task_config['mode'] = 'create'
```

Comment:
This seems like needing rethought.

(a) Its a bit magical that existence of schema distinguishes between create and append.
(b) In general pipelines will run multiple times, so things that are one-time only, like creation need to handle the case where the table already exists.

So I think a better behavior is to always create if it doesn't exist, and always require the schema.

Reply:
Here, I'm trying to infer what the 'mode' should be when just schema/table/path are present/absent. 

@nikhilk : If the table already exists, why would the schema be required?